documentation/config_file_auth_migration.md

```diff
@@ -0,0 +1,267 @@
+---
+layout: default
+title: "Puppet Server Configuration Files: Migrating to the HOCON auth.conf format"
+canonical: "/puppetserver/latest/config_file_auth_migration.html"
+---
+
+[Puppet `auth.conf`]: /puppet/latest/reference/config_file_auth.html
+[deprecated]: ./deprecated_features.html
+
+Puppet Server 2.2.0 introduces a significant change in how it manages authentication to API endpoints. The older [Puppet `auth.conf`][] file and whitelist-based authorization method are [deprecated][]. Puppet Server's new `auth.conf` file, illustrated below in examples, also uses a different format for authorization rules.
+
+Use the following examples and methods to convert your authorization rules when upgrading from Puppet Server 2.2.0 and newer. For detailed information about using the new or deprecated `auth.conf` rules with Puppet Server, see the [Puppet Server `auth.conf` documentation](./config_file_auth.html).
+
+> **Note:** To continue using the deprecated Ruby [Puppet `auth.conf`][] file and authorization methods, see the [Deprecated Ruby Parameters](/puppet/latest/reference/config_file_auth.html#deprecated-ruby-parameters) section of the `auth.conf` documentation.
+
+## Managing rules with Puppet modules
+
+You can reimplement and manage your authorization rules in the new HOCON format and `auth.conf` file by using the [`puppetlabs-puppet_authorization`](https://forge.puppet.com/puppetlabs/puppet_authorization) Puppet module. See the module's documentation for details.
+
+## Converting rules directly
+
+Most of the deprecated authorization rules and settings are available in the new format.
+
+> ### Unavailable rules, settings, or values
+>
+> The following rules, settings, and values have no direct equivalent in the new HOCON format. If you require them, you must reimplement them differently in the new format.
+>
+> * **on value of `auth`:** The deprecated `auth` parameter's on value results in a match only when a request provides a client certificate. There is no equivalent behavior in the HOCON format.
```

Comment:
This might be too pedantic to describe here, not sure...

With the Ruby auth.conf format, the "auth" keyword was evaluated as part of the determination of whether the incoming request matches the corresponding rule or not.  In the Puppet Server tk-auth format, though, there is no methodology for expressing that the request matches the rule  based on whether the user was authenticated or not.  Instead, in the Puppet Server tk-auth format, once it is determined that a request matches a rule, the request will then be allowed / denied based on whether the user had authenticated as a matching user name or the user was not authenticated ("allow-unauthenticated: true").

What you have here may cover this sufficiently for what users need to know about this.  Just throwing this out in the event that it might lead you to want to reword this.